features/src/FeatureContext.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Behat\MinkExtension\Tests;
+
+use Behat\Mink\Exception\ExpectationException;
+use Behat\MinkExtension\Context\RawMinkContext;
+
+/**
+ * Feature context for testing advanced scenarios.
+ */
+class FeatureContext extends RawMinkContext
+{
+
+    /**
+     * @BeforeScenario @MockXdebug
+     */
+    public function setUpXdebugMock()
+    {
+        $_SERVER['XDEBUG_CONFIG'] = 'xdebug';
+    }
+
+    /**
+     * @Then /^I should have the "([^"]*)" cookie with value "([^"]*)"$/
+     */
+    public function iShouldHaveTheCookieWithValue($cookie_name, $cookie_expected_value) {
+        if ($cookie_real_value = $this->getSession()->getCookie($cookie_name)) {
+            if ($cookie_real_value !== $cookie_expected_value) {
+                throw new ExpectationException(
+                  'The cookie with name ' . $cookie_name . ' was found, but does not contain ' . $cookie_real_value . ', yet it contains ' . $cookie_expected_value . '.',
+                  $this->getSession()
```

Comment:
passing the session in the constructor is deprecated. Please use the new API.

Btw, you could use the Mink assertion API for this: `$this->assertSession()->cookieEquals(...)` instead of duplicating its logic